[Tan Wei En] iP

tanweien opened the pull request:
## Duke
> If you wish, you may write the PR description to be very similar to the example given above -- as the goal here is to demonstrate your mastery of the GFMD syntax (not ad writing skills) - Prof

DukePro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://www.ntu.edu.sg).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 👍 

And it is FREE!
Features:

 - [x] Managing tasks
 - [ ] Managing deadlines (coming soon)
 - [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

tanweien committed:
Level 0, to test commit

tanweien committed:
Level 1

tanweien committed:
Level 2, added TaskList class

tanweien committed:
Level 2, added TaskList class

tanweien committed:
Level 4, added subclasses and fixed bugs

tanweien committed:
Level 5, added exception handling for tasks and markings

tanweien committed:
Level 6, added delete in Duke and TaskList

tanweien committed:
Level 6.1, cleaned up code before starting on other increments

tanweien committed:
Level 7, testing branching

tanweien committed:
Level 7, added save and fixed bugs

tanweien committed:
Level 8, added DateTime format

tanweien committed:
Level 8, merging branches

tanweien committed:
Level 8.1, merging branches

tanweien committed:
Level 8.2, update gitignore

tanweien committed:
Level 8.2, update gitignore

tanweien committed:
Level 9.0, adding more OOP

tanweien committed:
Level 9.1 cleaned up OOP bugs

tanweien committed:
Level A-Packages, organised into packages

tanweien committed:
Level A-Packages, organised into packages

tanweien committed:
A-Jar, packaged app as an executable JAR file

tanweien committed:
A-JavaDoc, included JavaDoc comments to the code

tanweien committed:
A-CodingStandard

tanweien committed:
Level-9, included find functionality

tanweien committed:
Merge branch 'branch-A-CodingStandard'

tanweien committed:
JavaDoc

tanweien committed:
Add Parser

zunedz commented on src/main/java/duke/AddCommand.java:
```diff
@@ -0,0 +1,24 @@
+    }
+
+    @Override
+    boolean isExit() {
```
very good that you put "is" prefix since this method returns a boolean

zunedz commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,40 @@
+        this.date = date;
+    }
+
+    public Deadline(String task, boolean complete, LocalDate date) {
```
Should the variable complete have a prefix (isComplete, hasCompleted) since it is a boolean variable?

zunedz commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,41 @@
+        this.date = date;
+    }
+
+    public Event(String task, boolean complete, LocalDate date) {
```
I think variable complete also have the same issue with the other one.

zunedz commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,52 @@
+abstract class Task {
+    private final String task;
+    private final String type;
+    private boolean completed;
```
I suppose this also need to add prefix?

zunedz commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,52 @@
+        return this.task;
+    }
+
+    public void setCompleted() {
```
great! adhere to the naming convention for methods by using verbs in the name.

zunedz commented on src/test/java/duke/DukeTest.java:
```diff
@@ -0,0 +1,17 @@
+    }
+
+    @Test
+    public void testRun() {
```
maybe you can consider this in naming test methods?
> Underscores may be used in test method names using the following three part format featureUnderTest_testScenario_expectedBehavior()

fredtwt commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,90 @@
+        } else if (ui.getCommandWord(fullCommand).equals("event")) {
+            if (ui.isValidTask(fullCommand)) {
+                if (ui.isValidEvent(fullCommand)) {
+                    try {
+                        String taskName = ui.getTaskName(fullCommand);
+                        LocalDate localDate = ui.getTaskDate(fullCommand);
+                        Event event = new Event(taskName, localDate);
+                        AddCommand addCommand = new AddCommand("event", event);
+                        return addCommand;
+                    } catch (DateTimeParseException e) {
+                        ui.showDateTimeParseException();
+                    }
+                } else {
+                    ui.showInvalidEvent();
+                }
+            } else {
+                ui.showEmptyTask();
+            }
+        } else if (ui.getCommandWord(fullCommand).equals("mark")) {
+            if (ui.isValidMark(fullCommand)) {
+                MarkCommand markCommand = new MarkCommand(fullCommand,
+                        ui.markIndex(fullCommand));
+                return markCommand;
+            }
+        } else if (ui.getCommandWord(fullCommand).equals("unmark")) {
+            if (ui.isValidUnmark(fullCommand)) {
+                UnmarkCommand unmarkCommand = new UnmarkCommand(fullCommand,
+                        ui.markIndex(fullCommand));
+                return unmarkCommand;
+            }
+        } else if (ui.getCommandWord(fullCommand).equals("find")) {
+            if (ui.isValidTask(fullCommand)) {
+                FindCommand findCommand = new FindCommand(fullCommand);
+                return findCommand;
+            } else {
+                ui.showGeneralException();
+            }
+        }
+        return null;
+    }
```
Could you truncate this method into multiple shorter methods instead?

fredtwt commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,225 @@
+                return true;
+            }
+        } else if (getCommandWord(fullCommand).equals("deadline")) {
+            if (fullCommand.indexOf("deadline") + 9 >= fullCommand.indexOf("/by")) {
```
Should you use a named constant instead for the magic number?

fredtwt reviewed (commented):
Overall, i think it's a great effort on maintaining the code quality. Apart for that one method that was over 30 LOC, you have a tendency to use magic numbers which can easily be resolved so work on that! Otherwise, really good job! 👍

tanweien committed:
Debug code

tanweien committed:
Improve OOP

tanweien committed:
no message

tanweien committed:
Merge branch 'add-gradle-support' of https://github.com/tanweien/ip into A-Gradle

tanweien committed:
Add Branch A-Gradle

tanweien committed:
Add checkstyle

tanweien committed:
Add Gui

tanweien committed:
Add Gui, fix bugs in code

tanweien committed:
Add assertions

tanweien committed:
Add assertions

tanweien committed:
Merge pull request #2 from tanweien/branch-A-CodeQuality

Improve code quality

tanweien committed:
Add assertions

tanweien committed:
Add sort

tanweien committed:
Fix gui bugs

tanweien committed:
Improve GUI functionality

tanweien committed:
Improves on Javadocs for Duke

tanweien committed:
Improves GUI

tanweien committed:
Deadline and event: wrong date format does not throw exception

Date format is valid

Proper date format is imperative for parser to add new deadline and event

Let's,
* Update methods to throw invalid datetime exception

tanweien committed:
Deadline and event: insufficient junit test

Junit test is implemented

Proper testing imperative to reduce bugs

Let's,
* Implement junit test for deadline and event instantiation

tanweien committed:
Improve Duke GUI

tanweien committed:
Add product screenshot

tanweien committed:
Add user guide

tanweien committed:
Improve on quality of code

tanweien committed:
Improve UI

tanweien committed:
Improve UI

tanweien committed:
Improve tasklist functionality
